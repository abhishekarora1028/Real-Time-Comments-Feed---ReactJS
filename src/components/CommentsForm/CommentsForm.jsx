import React, { useState } from 'react';
import { Form, Button, Row, Col } from 'react-bootstrap';
import socketIOClient from "socket.io-client";
import {store} from 'react-notifications-component';
import { CommentService } from '../../services/CommentService';
import loader from '../../assets/loading.svg';
import './CommentsForm.scss';

const socket = socketIOClient('http://localhost:3001');
const CommentsForm = (props) => {

    const [values, setValues] = useState({
        name: '',
        message: '',
        response:''
    });
    const [response, setResponse] = useState('');
    const [loading, setLoading] = useState(false);

    const handleSubmit = () => {
        setResponse('');
        let obj = JSON.parse(JSON.stringify(values));
        setLoading(true);
        try {
            CommentService.post('/createComment', obj).then(() => {
                setValues({
                    name: '',
                    message: '',
                    comment:'',
                })
                socket.emit("new_comment");
                store.addNotification({
                  title: "New Comment!",
                  message: "A new comment just got posted!",
                  type: "success",
                  insert: "top",
                  container: "bottom-left",
                  animationIn: ["animate__animated", "animate__fadeIn"],
                  animationOut: ["animate__animated", "animate__fadeOut"],
                  dismiss: {
                    duration: 3000,
                    onScreen: true
                  }
                });
            });
        } catch (err) {
            console.log(err);
        }
        setResponse('Comment posted successfully...');
        window.setTimeout(()=>{setResponse('');}, 3000)
        setLoading(false);
    }

    
    const changeValues = (value, key) => {
        let data = JSON.parse(JSON.stringify(values));
        data[key] = value;
        setValues(data);
    }

    return (
        <Form data-testid="commentsform">
            <Row>
                <Col sm={12} className="py-2">
                    {response !=="" ? 
                    <div className="alert alert-success" role="alert">{response}</div>
                    : "" }
                </Col>
            </Row>
            <Row>
            <Col sm={12}>
            <Form.Group controlId="formBasicEmail">
                <Form.Label>Name</Form.Label>
                <Form.Control type="text" name="name" value={values.name}  required onChange={(e) => changeValues(e.target.value, 'name')}  placeholder="Your name goes here..." />
            </Form.Group>
            </Col>
            </Row>
            <Row>
            <Col sm={12}>
            <Form.Group controlId="formBasicEmail">
                <Form.Label>Comment</Form.Label>
                <Form.Control as="textarea" rows="4" value={values.message} name="message" required  onChange={(e) => changeValues(e.target.value, 'message')} placeholder="Write your comment message here..." />
            </Form.Group>
            </Col>
            </Row>
            <Row>
            <Col sm={12} className="d-flex flex-row-reverse">
            <Button variant="primary" data-testid="commentbutton" disabled={loading} className="btn btn-primary" onClick={() => handleSubmit()}>
                        {loading ? <img style={{ width: '10%' }} src={loader} alt='loading' /> : 'Comment'}
            </Button>
            </Col>
            </Row>
      </Form>
    );
}

export default CommentsForm;