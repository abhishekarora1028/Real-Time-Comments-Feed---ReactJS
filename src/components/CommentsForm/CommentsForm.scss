.form-control {
    height: 42px;
    background-color: var(--light);
    border: 0;
    box-shadow: -1px 2px 3px var(--extra-light);
    padding: 5px 10px;
    font-size: 13px;
}