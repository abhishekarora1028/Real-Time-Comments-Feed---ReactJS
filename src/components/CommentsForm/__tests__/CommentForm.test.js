import React from 'React';
import { render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CommentsForm from '../CommentsForm';

afterEach(cleanup);

describe('CommentForm', () => {

    it('Should render the comment form successfully', () =>{
        const { asFragment } = render(<CommentsForm/>);
        expect(asFragment(<CommentsForm/>)).toMatchSnapshot();
    });
});