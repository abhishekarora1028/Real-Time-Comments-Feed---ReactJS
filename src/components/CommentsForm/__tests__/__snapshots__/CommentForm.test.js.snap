// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm Should render the comment form successfully 1`] = `
<DocumentFragment>
  <form
    class=""
    data-testid="commentsform"
  >
    <div
      class="row"
    >
      <div
        class="py-2 col-sm-12"
      />
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-12"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="formBasicEmail"
          >
            Name
          </label>
          <input
            class="form-control"
            id="formBasicEmail"
            name="name"
            placeholder="Your name goes here..."
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-12"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="formBasicEmail"
          >
            Comment
          </label>
          <textarea
            class="form-control"
            id="formBasicEmail"
            name="message"
            placeholder="Write your comment message here..."
            required=""
            rows="4"
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="d-flex flex-row-reverse col-sm-12"
      >
        <button
          class="btn btn-primary btn btn-primary"
          data-testid="commentbutton"
          type="button"
        >
          Comment
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
